### ISSUE TITLE
Logger hard to use

### ISSUE THREAD

**Issue by ichorid** [Tribler/tribler#5340]
Our logger now is hard to use. It should be configured much easier, reconfigurable by the user, break in case of big modules refactorings.

**Comment by devos50** [comment]
Is this issue addressed sufficiently now that we moved to `pyyaml`-based configurations?

**Comment by ichorid** [comment]
No, it is still not. The user still can't configure the thing on their own, because Tribler looks for `yaml`-based log config only in the source dirs. The user-configurable `yaml` - based logger config must be applied *after* the default one, and it should only contain *changed* entries. Otherwise, it is too wordy.

**Comment by devos50** [comment]
Ah I see. I actually wonder if users would use this feature. The log is not something that users are going to investigate. Instead, they just send their stack traces to us, and we debug by inspecting the logs, adding more logging statements if necessary.

At best, this would be a nice feature for developers but I think they can as easily just modify the main `logger.yaml` file.

**Comment by ichorid** [comment]
This is a really important feature for developers, as it allows us to enable logging in *compiled* Tribler packages.

**Comment by xoriole** [comment]
Actually, logging is not working at all. I don't see any logs being written to the log file.
Also the reason why logs section in the debug pane.
<img width="1237" alt="Screen Shot 2020-10-30 at 10 15 52" src="https://user-images.githubusercontent.com/1442867/97684067-f30d0500-1a98-11eb-9a71-d6a58fde2a13.png">

**Comment by qstokkink** [comment]
The logger has been reworked in the latest Tribler code and this issue is no longer relevant.
